templates/pkg/resource/manager.go.tpl

```diff
@@ -234,6 +239,17 @@ func (rm *resourceManager) lateInitializeFromReadOneOutput(
 {{ GoCodeLateInitializeFromReadOne .CRD "observed" "latest" 1 }}
 }
 
+// IsSynced returns true if the resource is synced.
+func (rm *resourceManager) IsSynced(ctx context.Context, res acktypes.AWSResource) (bool, error) {
+	r := rm.concreteResource(res)
+	if r.ko == nil {
+		// Should never happen... if it does, it's buggy code.
+		panic("resource manager's IsSynced() method received resource with nil CR object")
+	}
+{{ GoCodeIsSynced .CRD "r.ko" 1}}
+	return false, nil
```

Comment:
How about default being `true,nil`

I suggest this because I can see this method being called after successful `create/update + late-initialize` call to determine the synced status. In case of majority of AWS resources which have synchronous create operations, this default true will set Synced condition correctly and they will not need to explicitly add Synced conditions in their generator.yaml file.

Only issue with above approach will be the asynchronous create/update calls, which are minority of total AWS resources BTW. For those resources service teams can get rid of their Synced Condition hooks and add synced Config instead.

What do you think?

Reply:
>this default true will set Synced condition correctly and they will not need to explicitly add Synced conditions in their generator.yaml file.

This makes a lot of sense actually. I can make the `return false, nil` part of the `GoCodeIsSynced` generation